### PR TITLE
Add a vtk reader tailored for Shamrock dumps.

### PR DESCRIPTION
**Pull request by y-lapeyre**
Shamrock is a new GPU hydrodynamics/magnetohydrodynamics code for astrophysical simulations. It was presented at the 5th Phantom Workshop and will be made public later this year.

### CODE REVIEW

**Review by ttricco — CHANGES_REQUESTED** [review]
1. read_header() and read_vtk_series() are not used. I recommend removing this code, or do you plan to make use of this in future versions of Shamrock?

2. My inclination would be to rename this to read_shamrock(), as I would expect read_vtk() to be a general function to handle any .vtk file (which is not the case).

3. Does the .vtk file not contain the pmass value? It would be better to not have pmass as a function argument if it can be read from the file.

File: sarracen/readers/read_shamrock.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        else:
+            df[array_name + 'x'] = numpy_array[:, 0]
+            df[array_name + 'y'] = numpy_array[:, 1]
+            df[array_name + 'z'] = numpy_array[:, 2]
+
+    # add B
+    df['Bx'] = df['B/rhox'] * df['rho']
+    df['By'] = df['B/rhoy'] * df['rho']
+    df['Bz'] = df['B/rhoz'] * df['rho']
```
Comment (ttricco) [inline sarracen/readers/read_shamrock.py:57]: Sorry just another comment I should have realized before now -- what if the file doesn't contain magnetic field? I can imagine that you would often pure hydro calculations.

A simple check would be:
`if 'B/rhox' in df.columns:`
`    df['Bx'] = df['B/rhox'] * df['rho']`

**Review by ttricco — CHANGES_REQUESTED** [review]

**Comment by ttricco** [comment]
I have renamed read_vtk() to read_shamrock() and will merge this in. Thanks for your contribution. Please raise an issue or open a new PR if any changes are needed.

**Comment by y-lapeyre** [comment]
Sorry for my late response, was out of internet for a week. Thank you very much! And unfortunately the VTK format doesn't allow for much metadata to be stored. According to the creators of the format, if one wants to store constants in the header, they should use an extra column for it... which is not ideal. We will come up with a Shamrock native format in the future.

**Pull request merged**